### PR TITLE
[FIX] pos_cash_rounding: fix "almost zero" error

### PR DESCRIPTION
**Pull request by yelizariev**
TLDR: zero is not always zero in js

STEPS:

* On your POS, add a Cash rounding > Rounding precision 0.10 > Rounding method
DOWN
* Set a product with a Unit of Measure with 3 decimals (kg for instance)
* In POS:

  - Make an order for 0.514 kg at 429.99€ + 0.510 kg at 429.99 + 0.60€
  - The total of the order is 440.90 (exact total would be 440.90976)

BEFORE: The total to pay is 440.80

AFTER:  The total to pay is 440.90

WHY:

* var total = round_pr(this.get_total_with_tax(), this.pos.cash_rounding[0].rounding);

  // 440.90000000000003

* var rounding_applied = total - this.get_total_with_tax();

  // 5.684341886080802e-14

---

opw-2357917

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by yelizariev** [comment]
@pimodoo @nim-odoo

File: addons/pos_cash_rounding/static/src/js/pos_cash_rounding.js [hunk 1]
```diff
@@ -43,7 +43,9 @@ models.Order = models.Order.extend({
 
             var rounding_applied = total - this.get_total_with_tax();
             // because floor and ceil doesn't include decimals in calculation, we reuse the value of the half-up and adapt it.
-            if(this.pos.cash_rounding[0].rounding_method === "UP" && rounding_applied < 0) {
+            if (utils.float_is_zero(rounding_applied)){
+                // https://xkcd.com/217/
```
Comment (nim-odoo) [inline addons/pos_cash_rounding/static/src/js/pos_cash_rounding.js:47]: This would be a bit clearer I think:
```suggestion
                // https://xkcd.com/217/
                return 0;
```

**Review by nim-odoo — APPROVED** [review]
LGTM, let's see if it's ok for @pimodoo

**Comment by pimodoo** [comment]
@robodoo r+ rebase-ff
